Update form 21P-0537: content updates and structural changes

amponce opened the pull request:
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder


Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

Content feedback and confirm page:

- Update form title to "Verify your marital status for DIC benefits"
- Update subtitle to "Marital Status Questionnaire (VA Form 21P-0537)"
- Convert IntroductionPage to use IntroductionPageView component
- Simplify introduction page content per content feedback
- Update SaveInProgressIntro startText to "Verify your marital status"

- Move phone and email to first chapter (Your contact information)
- Update contact info to use primary/secondary phone pattern
- Remove marriage recognition page from flow
- Remove marriage recognition alert from review page
- Update chapter title to "Remarriage information"

- Update radio questions to use labelHeaderLevel for h3 heading behavior
- Update marriage wording: "marriage" to "remarriage", "spouse" to "spouse's"
- Update Deceased Veteran's identification page title and description
- Update Spouse's identification page to match Veteran pattern with hint text
- Update submit button text to "Submit"

- Convert ConfirmationPage to use ConfirmationView with ChapterSectionCollection
- Update prefill transformer to use correct user profile path
- Update submit transformer to handle new phone/email field names
- Update all test fixtures to use new data structure (primaryPhone, secondaryPhone, emailAddress)


## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#0000


## Testing done

- Unit tests are in a separate PR, verified changes manually and ran existing tests


## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

| Review and Submit | Confirmation |
|-------------------|--------------|
| <img width="1298" height="4775" alt="localhost_3001_supporting-forms-for-claims_marital-status-questionnaire-21p-0537_contact_phone-email" src="https://github.com/user-attachments/assets/8e043ec1-15f9-4234-8d66-c252f9accf78" /> | <img width="1298" height="3105" alt="localhost_3001_supporting-forms-for-claims_marital-status-questionnaire-21p-0537_confirmation" src="https://github.com/user-attachments/assets/bbfe42a2-4b03-4599-8af0-02683865dc23" /> |


## What areas of the site does it impact?

[marital-status-questionnaire-](http://localhost:3001/supporting-forms-for-claims/marital-status-questionnaire-21p-0537)

## Acceptance criteria

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

amponce committed:
Update form 21P-0537: content updates and structural changes

- Update form title to "Verify your marital status for DIC benefits"
- Update subtitle to "Marital Status Questionnaire (VA Form 21P-0537)"
- Convert IntroductionPage to use IntroductionPageView component
- Simplify introduction page content per content feedback
- Update SaveInProgressIntro startText to "Verify your marital status"

- Move phone and email to first chapter (Your contact information)
- Update contact info to use primary/secondary phone pattern
- Remove marriage recognition page from flow
- Remove marriage recognition alert from review page
- Update chapter title to "Remarriage information"

- Update radio questions to use labelHeaderLevel for h3 heading behavior
- Update marriage wording: "marriage" to "remarriage", "spouse" to "spouse's"
- Update Deceased Veteran's identification page title and description
- Update Spouse's identification page to match Veteran pattern with hint text
- Update submit button text to "Submit"

- Convert ConfirmationPage to use ConfirmationView with ChapterSectionCollection
- Update prefill transformer to use correct user profile path
- Update submit transformer to handle new phone/email field names
- Update all test fixtures to use new data structure (primaryPhone, secondaryPhone, emailAddress)

michaelclement commented on src/applications/simple-forms/21p-0537/pages/phoneAndEmail.js:
```diff
@@ -1,47 +1,23 @@
-      },
+      primaryPhone: phoneSchema,
+      secondaryPhone: phoneSchema,
+      emailAddress: emailToSendNotificationsSchema,
```
One thought - the PDF limits emails to 30 characters, so if we added some validation here we could hopefully prevent that from being an issue for users:
```javascript
      emailAddress: {
        ...emailToSendNotificationsSchema,
        maxLength: 30,
      },
```

michaelclement reviewed (dismissed):
Looks great - tested a few submissions out locally and worked as expected. Not a blocker, but if you want to add the email charcount in I can do a quick re-approve 👍

amponce committed:
Code review feedback for email char count

amponce committed:
Fix unit tests for content changes

- Update ConfirmationPage tests for ConfirmationView component
- Update IntroductionPage tests to match new content and OMB date
- Fix prefill transformer tests to match actual field names
- Add tests for new pages: phoneAndEmail, remarriageQuestion, terminationStatus
- Remove obsolete marriageRecognition page tests
- Update form config tests to remove non-existent page references

michaelclement reviewed (approved):
Looks good!
